rbac/management/principal/model.py

```diff
@@ -26,14 +26,19 @@
 class Principal(TenantAwareModel):
     """A principal."""
 
+    class Types(models.TextChoices):
+        USER = "user", "User"
+        SERVICE_ACCOUNT = "service-account", "Service Account"
```

Comment:
Sort of a nit, but we could probably trim the values down on these to be 1-3 characters, which probably doesn't do a ton for optimization, but might not hurt? Like `SERVICE_ACCOUNT = "SA", "Service Account"` for instance, and then we could have a `max_length` on the column definition below. 🤷 Again probably not a big deal, but since the type is effectively an enum with a human-readable counterpart it might be worth it?

Reply:
Also, with this since we don't use it in a form or anything, we could probably just get away with:

```python
USER = "user"
SERVICE_ACCOUNT = "service-account"
```
We'd still have the `label` and `name` attributes based off the constant name:
```python
>>> Types.USER.__dict__
{'_value_': 'user', '_name_': 'USER', '__objclass__': <enum 'Types'>, '_label_': 'User'}

>>> Types.SERVICE_ACCOUNT.__dict__
{'_value_': 'service-account', '_name_': 'SERVICE_ACCOUNT', '__objclass__': <enum 'Types'>, '_label_': 'Service Account'}
```

Reply:
Yea, we still need db_index to optimize the query. The type is existing data, so I am using service-account and user. We need a migration to migrate existing data before we can use `SA` though, I will add a JIRA task for this to consider more. At least add a db_index to the field to help optimize the query speed

- Being explicit can help reading the code and db records; 
- The short form can help reducing some db space;